inc/dbmysql.class.php

```diff
@@ -1037,6 +1037,25 @@ function updateOrDie($table, $params, $where, $message = '') {
       return $res;
    }
 
+   /**
+    * Update a row in the database or insert a new one
+    *
+    * @since 9.4
+    *
+    * @param string $table   Table name
+    * @param array  $params  Query parameters ([:field name => field value)
+    * @param array  $where   WHERE clause
+    *
+    * @return mysqli_result|boolean Query result handler
+    */
+   function updateOrInsert($table, $params, $where) {
+      $res = $this->update($table, $params, $where);
+      if ($this->affected_rows() == 0) {
```

Comment:
Are you sure `affected_rows` is not equal to 0 if record exists but ha nos been changed?

Reply:
Indded, equal to 0 if no changes. We have to specify the CLIENT_FOUND_ROWS flag at mysqli connection to get it work (affected rows = matched rows after that). But it's a too important modification for this little feature.

I'll keep searching

Reply:
Changing mysqli connection flag is indeed a bit too muc and should impact many other features :/

Reply:
This need to be confirmed, but `update()` method apparently returns `false` when something went wrong, not if `affected_rows == 0`.
Maybe just a `if ($res !== false) {}` would do the job.